wire-ios-system/Source/Logging/WireLogger+Datadog.swift

```diff
@@ -0,0 +1,27 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+extension WireLogger {
+    public static func addDatadog(_ datadog: LoggerProtocol) {
+        if let aggregatedLogger = Self.provider as? AggregatedLogger {
+            aggregatedLogger.addLogger(datadog)
+        } else {
+            Self.provider = datadog
```

Comment:
I haven't checked where `provider` is used, but just from seeing this line here I wonder if it shouldn't try to not replace the existing provider but again use the `AggregatedLogger`, somehow like
```suggestion
            Self.provider = AggregatedLogger(loggers: [Self.provider, datadog])
```
Maybe a check is needed if the current provider isn't already a or the datadog logger.

Reply:
thanks but will not touch it in this PR as the code was just moved